lib/autoupdate/agent/telemetry.go

```diff
@@ -103,3 +108,38 @@ func hasParentDir(dir, parent string) (bool, error) {
 	}
 	return strings.HasPrefix(absDir, absParent), nil
 }
+
+type UpdaterInfo struct {
+	Group  string
+	UUID   uuid.UUID
+	Status proto.UpdaterStatus
+}
+
+func HelloUpdaterInfo() (*proto.UpdaterV2Info, error) {
```

Comment:
```suggestion
func ReadHelloUpdaterInfo() (*proto.UpdaterV2Info, error) {
```

I'd also be a little wary of mixing IO and logic like this, I'd prefer if it was a function that converted a `*UpdateConfig` into a `*proto.UpdaterV2Info` or something.

Reply:
(also godocs)